### PR TITLE
Add documentation for why diverging from the spec at AASd-116

### PR DESCRIPTION
**Pull request by s-heppner**
(case insensitive checking of string) 
See #205

AASd-116 
> "globalAssetId" (case-insensitive) is a reserved key. If used as value for SpecificAssetId/name, SpecificAssetId/value shall be identical to AssetInformation/globalAssetId.
> Note: AASd-116 is important to enable a generic search across global and specific asset IDs.

(Chapter 5.3.12.5, page 112)
https://nextcloud.s-heppner.com/index.php/s/2KZr2kHoSTjkSoi#page=112

AASd-116 asks for case-insensitive checking of the key "globalAssetId". We diverge from this and write an explanation as to why.

### CODE REVIEW

**Review by g1zzm0 — CHANGES_REQUESTED** [review]
Wording doesn't match latest Version of Spec

**Comment by s-heppner** [comment]
We diverge from the wording and the implementation meaningfully, which is why we added the notes in the docstrings

File: aas_core_meta/v3.py [hunk 1]
```diff
@@ -24,6 +24,10 @@
 
 * :constraintref:`AASd-012`
 
+:constraintref:`AASd-116`: In this constraint the string ``globalAssetId`` is checked case-sensitively,
```
Comment (mristin) [inline aas_core_meta/v3.py:27]:
```suggestion
* :constraintref:`AASd-116`: In this constraint the string ``globalAssetId`` is checked case-sensitively,
```

@s-heppner I think this is also an item in the preceding bullet list ("language understanding")?

Comment (mristin) [inline aas_core_meta/v3.py:27]: @s-heppner It is not clear from the text what the book says and where we diverge.

Suggestion: "In the book, :constraintref:`AASd-116` imposes a case-*insensitive* equality against ``globalAssetId``. This is culturally-dependent, and depends on the system settings. For example, the case-folding for the letter "i" and "I" is different from English.

We implement the constraint as case-*sensitive* instead to allow for interoperability across different culture settings.

Comment (mristin) [reply aas_core_meta/v3.py:27]: @s-heppner this constraint could also be approximated as `[gG][lL][oO]...`. Is that not an option? Obviously, this works only for some languages.

Comment (s-heppner) [reply aas_core_meta/v3.py:27]: Both ideas are really good, imo. I'll deliberate with the team which we would want to choose.

Comment (mristin) [reply aas_core_meta/v3.py:27]: 👍

Comment (s-heppner) [reply aas_core_meta/v3.py:27]: We decided, choosing your first suggested approach.

Comment (mristin) [reply aas_core_meta/v3.py:27]: Fixed the missing part of the sentence:

 "In the book, :constraintref:`AASd-116` imposes a case-_insensitive_ equality against `globalAssetId`. This is culturally-dependent, and depends on the system settings. For example, the case-folding for the letters "i" and "I" is different in Turkish from English.

We implement the constraint as case-_sensitive_ instead to allow for interoperability across different culture settings."

File: aas_core_meta/v3.py [hunk 2]
```diff
@@ -24,6 +24,14 @@
 
 * :constraintref:`AASd-012`
 
+:constraintref:`AASd-116`: In the book, :constraintref:AASd-116 imposes a
+case-insensitive equality against globalAssetId. This is culturally-dependent,
+and depends on the system settings. For example, the case-folding
+for the letters "i" and "I" is different in Turkish from English.
+
+We implement the constraint as case-sensitive instead to allow for interoperability
+across different culture settings.
```
Comment (mristin) [inline aas_core_meta/v3.py:33]:
```suggestion
* :constraintref:`AASd-116`: In the book, :constraintref:AASd-116 imposes a
  case-insensitive equality against globalAssetId. This is culturally-dependent,
  and depends on the system settings. For example, the case-folding
  for the letters "i" and "I" is different in Turkish from English.

  We implement the constraint as case-sensitive instead to allow for interoperability
  across different culture settings.
```

File: aas_core_meta/v3.py [hunk 3]
```diff
@@ -24,6 +24,14 @@
 
 * :constraintref:`AASd-012`
 
+:constraintref:`AASd-116`: In the book, :constraintref:AASd-116 imposes a
```
Comment (mristin) [inline aas_core_meta/v3.py:27]:
```suggestion
In the book, :constraintref:AASd-116 imposes a
```

File: aas_core_meta/v3.py [hunk 4]
```diff
@@ -24,6 +24,14 @@
 
 * :constraintref:`AASd-012`
 
+In the book, :constraintref:AASd-116 imposes a
+case-insensitive equality against globalAssetId. This is culturally-dependent,
+and depends on the system settings. For example, the case-folding
+for the letters "i" and "I" is different in Turkish from English.
+
+We implement the constraint as case-sensitive instead to allow for interoperability
+across different culture settings.
```
Comment (mristin) [inline aas_core_meta/v3.py:33]:
```suggestion

* :constraintref:`AASd-012`
* :constraintref:`AASd-116`: In the book, :constraintref:`AASd-116` imposes a
  case-insensitive equality against globalAssetId. This is culturally-dependent,
  and depends on the system settings. For example, the case-folding
  for the letters "i" and "I" is different in Turkish from English.

  We implement the constraint as case-sensitive instead to allow for interoperability
  across different culture settings.
```

File: aas_core_meta/v3.py [hunk 5]
```diff
@@ -2108,7 +2116,7 @@ def __init__(
             for specific_asset_ID in self.specific_asset_IDs
         )
     ),
-    "AASd-116: ``globalAssetId`` is a reserved key. "
+    "Constraint AASd-116: ``globalAssetId`` is a reserved key. "
     "If used as value for ``Specific_asset_ID.name`` then ``Specific_asset_ID.value`` "
     "shall be identical to ``global_asset_ID``."
```
Comment (mristin) [inline aas_core_meta/v3.py:2121]:
```suggestion
    "Constraint AASd-116: ``globalAssetId`` is a reserved key. "
    "If used as value for the name of specific asset ID then the value of specific asset ID "
    "shall be identical to the global asset ID."
```

File: aas_core_meta/v3.py [hunk 6]
```diff
@@ -2137,6 +2145,19 @@ class Asset_information(DBC):
 
         For AssetInformation either the :attr:`global_asset_ID` shall be defined
         or at least one specificAssetId.
+
+    .. note::
+
+        Constraint AASd-116 is important to enable a generic search across global and
```
Comment (mristin) [inline aas_core_meta/v3.py:2151]:
```suggestion
        :constraintref:`AASd-116` is important to enable a generic search across global and
```

File: aas_core_meta/v3.py [hunk 7]
```diff
@@ -2137,6 +2145,19 @@ class Asset_information(DBC):
...
+        :constraintref:`AASd-116` is important to enable a generic search across global and
+        specific asset IDs.
+
+    .. note::
+
+        We diverge from the specification, where ``globalAssetId`` is checked
+        case-insensitive, as it would be not well defined, which characters would be
+        equal in this kind of check. Therefore, here, the string ``globalAssetId``
+        is checked case-sensitively. For example, would the UTF-8 characters
+        U+04E7 and U+00F6 be equal?
```
Comment (mristin) [inline aas_core_meta/v3.py:2160]:
```suggestion
		In the book, :constraintref:`AASd-116` imposes a
		case-insensitive equality against globalAssetId. This is 
		culturally-dependent, and depends on the system settings. 
		For example, the case-folding for the letters "i" and "I" is 
		different in Turkish from English.

		We implement the constraint as case-sensitive instead to allow 
		for interoperability across different culture settings.
```

**Pull request merged**